elytron/src/main/java/org/wildfly/extension/elytron/DomainDefinition.java

```diff
@@ -193,6 +195,13 @@ class DomainDefinition extends SimpleResourceDefinition {
             .setCapabilityReference(SECURITY_DOMAIN_CAPABILITY, SECURITY_DOMAIN_CAPABILITY)
             .build();
 
+    static final StringListAttributeDefinition TRUSTED_VIRTUAL_SECURITY_DOMAINS = new StringListAttributeDefinition.Builder(ElytronDescriptionConstants.TRUSTED_VIRTUAL_SECURITY_DOMAINS)
+            .setRequired(false)
+            .setMinSize(1)
+            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .setCapabilityReference(VIRTUAL_SECURITY_DOMAIN_CAPABILITY, SECURITY_DOMAIN_CAPABILITY)
```

Comment:
The CLI autocomplete feature does not work correctly for this attribute. When there is a dot in the name of a virtual security domain just the extension is offered. For example

```
/subsystem=elytron/virtual-security-domain=abc.war:add
/subsystem=elytron/security-domain=ApplicationDomain:write-attribute(name=trusted-virtual-security-domains,value=[<TAB>
->
/subsystem=elytron/security-domain=ApplicationDomain:write-attribute(name=trusted-virtual-security-domains,value=[war
```
It works correctly for names without a dot.

Reply:
The problem is related to capabilities (CLI completion is based on that). The capability-registry returns the last item after the last dot (https://github.com/wildfly/wildfly-core/blob/main/controller/src/main/java/org/jboss/as/controller/CapabilityRegistry.java#L1034). For example, I created a security domain named foo.war. Then call:
/core-service=capability-registry:suggest-capabilities(name=org.wildfly.security.security-domain, dependent-address=[("subsystem" => "elytron"), ("http-authentication-factory" => "test") ])
You will see that the returned list only contains war.
I am wandering if that would be enough to returns the part of the capability name after the provided capability name. For example: "org.wildfly.security.security-domain" is the provided capability name, "org.wildfly.security.security-domain.foo.war" is the capability of a registration point that matches, we would return "foo.war". Perhaps @bstansberry could provide input there.

Reply:
Actually we have a conflicting case (capability "org.wildfly.undertow.host") with the CLI command: /subsystem=undertow:write-attribute(name=default-virtual-host,value=TAB that returns the capability "org.wildfly.undertow.host.default-server.default-host", in this case the prefix "default-server." must be removed. That was the driver to return only the last part after the dot (See https://issues.redhat.com/browse/WFCORE-2690)